### PR TITLE
Refactoring/128 change release strategy for scala repositories

### LINKED ISSUE #127: Add help command line option
### Situation

At the moment, release-droid does not recognize the `-h` or `--help` commands. It shows available commands, but also terminates with an unrecognized option exception.

```
$ java -jar ../release-droid/target/release-droid-0.3.1.jar --help
usage: Release Droid
 -b,--branch <arg>      git branch (only for validation)
 -g,--goal <arg>        goal to execute
 -l,--local <arg>       local path to the repository
 -n,--name <arg>        project name
 -p,--platforms <arg>   comma-separated list of release platforms
Exception in thread "main" java.lang.IllegalArgumentException: E-RR-RUN-1: Unrecognized option: --help
        at com.exasol.releasedroid.main.UserInputParser.getCommandLine(UserInputParser.java:59)
        at com.exasol.releasedroid.main.UserInputParser.parseUserInput(UserInputParser.java:27)
        at com.exasol.releasedroid.main.Runner.main(Runner.java:28)
```

### Acceptance Criteria

- `-h` and `--help` commands are added that show usage

### LINKED ISSUE #128: Change release strategy for Scala releases
In the previous commit we only refactored code for Java repositories

### PR DESCRIPTION
**Pull request by AnastasiiaSergienko**
Closes #128 
Closes #127

### CODE REVIEW

File: doc/user_guide/templates/prepare_original_checksum_template.md [hunk 1]
```diff
@@ -0,0 +1,39 @@
...
+        run: sbt assembly
+      - name: Running tests
+        run: sbt test it:test
+      - name: Prepare checksum
+        run: find target/scala*/stripped -name *.jar -exec sha256sum "{}" + > original_checksum
+      - name: Upload checksum to the artifactory
+        uses: actions/upload-artifact@v2
+        with:
+          name: original_checksum
+          retention-days: 5
```
Comment (morazow) [inline doc/user_guide/templates/prepare_original_checksum_template.md:37]: Do we need this retention-days parameter? It seems like it removes the checksum.

File: doc/user_guide/templates/sbt_file_template.md [hunk 2]
```diff
@@ -0,0 +1,17 @@
...
+lazy val root =
+  project
+    .in(file("."))
+    .settings(moduleName := "<project-name>")
+    .settings(version := "<project-version>")
+    .settings(orgSettings)
+    .settings(buildSettings)
+    .settings(Settings.projectSettings(scalaVersion))
+    .settings(
+      resolvers ++= Dependencies.Resolvers)
```
Comment (morazow) [inline doc/user_guide/templates/sbt_file_template.md:15]:
```suggestion
    .settings(resolvers ++= Dependencies.Resolvers)
```

**Review by morazow — APPROVED** [review]
Minor suggestions and a question.

File: doc/user_guide/templates/prepare_original_checksum_template.md (hunk 1 above)
Comment (AnastasiiaSergienko) [reply doc/user_guide/templates/prepare_original_checksum_template.md:37]: We discussed it via Slack, so I keep the parameter

**Review by morazow — APPROVED** [review]

**Pull request merged**